lib/util/insightspublisher/index.js

```diff
@@ -59,19 +63,31 @@ class InsightsPublisher extends EventEmitter {
       }
     });
 
-    const self = this;
-
-    this.on('disconnected', function maybeReconnect(error) {
-      self._session = null;
-      if (error && self._reconnectAttemptsLeft > 0) {
-        self.emit('reconnecting');
-        reconnect(self, token, sdkName, sdkVersion, options);
-        return;
-      }
-      self.removeListener('disconnected', maybeReconnect);
+    this._readyToConnect.promise.then(({ roomSid, participantSid }) => {
+      const self = this;
```

Comment:
@makarandp0 This can move out of the promise resolver.

Reply:
not sure what you mean - we do need `maybeReconnect` inside the resolver because it references `roomSid, participantSid` which are available only after promise is resolved.

regarding `self` that is inside the resolver because outside code does not need it.